concept/type_/owns.rs

```diff
@@ -51,66 +52,63 @@ impl<'a> Owns<'a> {
         snapshot: &impl ReadableSnapshot,
         type_manager: &TypeManager,
     ) -> Result<bool, ConceptReadError> {
-        type_manager.get_owns_is_key(snapshot, self.clone())
+        type_manager.get_is_key(snapshot, self.clone().into_owned())
     }
 
-    pub fn is_unique(
+    pub fn get_constraint_abstract(
         &self,
         snapshot: &impl ReadableSnapshot,
         type_manager: &TypeManager,
-    ) -> Result<bool, ConceptReadError> {
-        type_manager.get_owns_is_unique(snapshot, self.clone())
+    ) -> Result<Option<CapabilityConstraint<Owns<'static>>>, ConceptReadError> {
+        type_manager.get_capability_abstract_constraint(snapshot, self.clone().into_owned())
     }
 
-    pub fn is_distinct(
+    pub fn get_constraints_distinct(
         &self,
         snapshot: &impl ReadableSnapshot,
         type_manager: &TypeManager,
-    ) -> Result<bool, ConceptReadError> {
-        type_manager.get_owns_is_distinct(snapshot, self.clone())
+    ) -> Result<HashSet<CapabilityConstraint<Owns<'static>>>, ConceptReadError> {
+        type_manager.get_owns_distinct_constraints(snapshot, self.clone().into_owned())
     }
 
-    pub fn get_constraint_regex(
+    pub fn is_distinct(
```

Comment:
Instead of `is_distinct` maybe `has_constraint_distinct`? To keep it consistent?

Reply:
It is consistent, you have multiple interfaces for different purposes:

* `get_constraint_abstract` returns an `abstract` constraint if exists.
* `is_abstract` (in impl Capability, that's why you don't see it here and it may confuse you) calls this getter and checks `is_some`.

* `get_constraints_abstract` returns a collection of `distinct` constraints. `abstract` constraint is stored only once (because it's scoped to only a single instance of a single type) for a type, while you can have multiple `distinct` constraints, that why it's `Option` vs `HashSet`.
* `is_distinct` does a job similar to `is_abstract`.

You call the latter when you're interested if it's just abstract or distinct overall. You ask for a constraint to get its description and source.